network/peer/tls_config.go

```diff
@@ -26,5 +36,29 @@ func TLSConfig(cert tls.Certificate, keyLogWriter io.Writer) *tls.Config {
 		InsecureSkipVerify: true, //#nosec G402
 		MinVersion:         tls.VersionTLS13,
 		KeyLogWriter:       keyLogWriter,
+		VerifyConnection:   ValidateRSACertificate,
+	}
+}
+
+// ValidateRSACertificate validates TLS certificates
+// with RSA public keys in the leaf of the certificate chain of the given connection state.
+func ValidateRSACertificate(cs tls.ConnectionState) error {
+	if len(cs.PeerCertificates) == 0 {
+		return ErrNoCertsSent
+	}
+
+	if cs.PeerCertificates[0] == nil {
+		return ErrEmptyCert
+	}
+
+	pk := cs.PeerCertificates[0].PublicKey
+
+	switch rsaKey := pk.(type) {
+	case *rsa.PublicKey:
+		return staking.ValidateRSAPublicKeyIsWellFormed(rsaKey)
```

Comment:
This is definitely me just being overly defensive, but should we attempt to handle the case that `rsaKey` is `nil` here?

Test would look like:
```golang
{
	description: "nil RSA key",
	input: func() tls.ConnectionState {
		key, err := rsa.GenerateKey(rand.Reader, 2048)
		require.NoError(t, err)

		x509CertWithNilPK := makeRSACertAndKey(t, key)
		x509CertWithNilPK.cert.PublicKey = (*rsa.PublicKey)(nil)
		return tls.ConnectionState{PeerCertificates: []*x509.Certificate{&x509CertWithNilPK.cert}}
	},
	expectedErr: peer.ErrEmptyPublicKey,
},
```

Reply:
Added test + check in `ValidateRSAPublicKeyIsWellFormed`.